ctru-rs/src/services/ndsp/wave.rs

```diff
@@ -0,0 +1,138 @@
+use ctru_sys::ndspChnWaveBufClear;
+
+use super::AudioFormat;
+use crate::linear::LinearAllocator;
+
+/// Base struct to represent audio wave data. This requires audio format information.
+pub struct WaveBuffer {
+    /// Buffer data. This data must be allocated on the LINEAR memory.
+    data: Box<[u8], LinearAllocator>,
+    audio_format: AudioFormat,
+    nsamples: usize, // We don't use the slice's length here because depending on the format it may vary
+                     // adpcm_data: AdpcmData, TODO: Requires research on how this format is handled.
+}
+
+/// Informational struct holding the raw audio data and playaback info. This corresponds to [ctru_sys::ndspWaveBuf]
+pub struct WaveInfo<'b> {
```

Comment:
Before closing up the example I had in mind a different system for WaveBuffers and WaveInfos, so that they could remain separate things (helping with clones etc). That didn't come to be because of the difficulties `ndsp` gives with lifetimes. I'll just merge them.